From 8715972bb64526ac3fb301b66d42a463a633e2b0 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Thu, 29 Sep 2011 22:16:47 -0400
Subject: [PATCH] Fix implicit function declaration warnings

Fix implicit declaration of vinagre_vnc_connection_set_depth_profile by
including the plugins/vnc/vinagre-vnc-connection.h header.

Fix gtk_widget_modify_bg implicit declaration warning by switching to
the non-deprecated gtk_widget_override_background_color().

Fix *many* vinagre_utils_* and vinagre_dirs_* implicit declarations by
generating a header (vinagre/vinagre-utils.h) from the vala source, and
including it in half the .c files in the source tree.
Since automake won't do it automatically, we have to write an explicit
rule to generate the header from vala. Add the generated header to
BUILT_SOURCES to ensure that it gets built before anything that might
include it. And add it to noinst_vinagreh_headers to make sure that it's
distributed with the tarball, so end users won't need valac to build
vinagre.

https://bugzilla.gnome.org/show_bug.cgi?id=660531
---
 Makefile.am                                   |   11 +++++++++++
 plugins/rdp/vinagre-rdp-tab.c                 |    1 +
 plugins/spice/vinagre-spice-connection.c      |    1 +
 plugins/spice/vinagre-spice-tab.c             |    1 +
 plugins/vnc/vinagre-vnc-connection.c          |    1 +
 plugins/vnc/vinagre-vnc-tab.c                 |    1 +
 vinagre/vinagre-bookmarks-migration.c         |    1 +
 vinagre/vinagre-bookmarks-ui.c                |    1 +
 vinagre/vinagre-bookmarks.c                   |    1 +
 vinagre/vinagre-cache-prefs.c                 |    1 +
 vinagre/vinagre-commands.c                    |    1 +
 vinagre/vinagre-connect.c                     |    1 +
 vinagre/vinagre-connection.c                  |    1 +
 vinagre/vinagre-notebook.c                    |    1 +
 vinagre/vinagre-options.c                     |    2 ++
 vinagre/vinagre-reverse-vnc-listener-dialog.c |    1 +
 vinagre/vinagre-reverse-vnc-listener.c        |    1 +
 vinagre/vinagre-ssh.c                         |    1 +
 vinagre/vinagre-tab.c                         |    5 +++--
 vinagre/vinagre-tube-handler.c                |    2 ++
 vinagre/vinagre-window.c                      |    1 +
 21 files changed, 35 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 6c6b90a..dee9394 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -51,7 +51,11 @@ vinagre_vinagre_LDADD = \
 	$(ssh_plugin) \
 	libvnc.la
 
+built_headers = \
+	vinagre/vinagre-utils.h
+
 noinst_vinagreh_headers = \
+	$(built_headers)
 	vinagre/if/ifaddrs.h \
 	vinagre/view/autoDrawer.h \
 	vinagre/view/drawer.h \
@@ -223,6 +227,13 @@ libvnc_la_SOURCES = \
 libvnc_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
 libvnc_la_LIBADD = $(VNC_LIBS)
 
+BUILT_SOURCES = \
+	$(built_headers)
+
+# manual rule to generate vinagre-utils.h, since automake won't automatically do it
+vinagre/vinagre-utils.h: vinagre/vinagre-dirs.vala vinagre/vinagre-utils.vala
+	$(VALAC) $(AM_VALAFLAGS) $(VALAFLAGS) --header=$@ -C $^
+
 iconthemedir = $(datadir)/icons/hicolor
 mimeicon16dir = $(iconthemedir)/16x16/mimetypes
 mimeicon22dir = $(iconthemedir)/22x22/mimetypes
diff --git a/plugins/rdp/vinagre-rdp-tab.c b/plugins/rdp/vinagre-rdp-tab.c
index d7647dd..b0074af 100644
--- a/plugins/rdp/vinagre-rdp-tab.c
+++ b/plugins/rdp/vinagre-rdp-tab.c
@@ -30,6 +30,7 @@
 
 #include "vinagre-rdp-tab.h"
 #include "vinagre-rdp-connection.h"
+#include "vinagre-utils.h"
 
 #define VINAGRE_RDP_TAB_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), VINAGRE_TYPE_RDP_TAB, VinagreRdpTabPrivate))
 
diff --git a/plugins/spice/vinagre-spice-connection.c b/plugins/spice/vinagre-spice-connection.c
index 869096c..be7e8d2 100644
--- a/plugins/spice/vinagre-spice-connection.c
+++ b/plugins/spice/vinagre-spice-connection.c
@@ -25,6 +25,7 @@
 #include <vinagre/vinagre-cache-prefs.h>
 
 #include "vinagre-spice-connection.h"
+#include "vinagre-utils.h"
 
 struct _VinagreSpiceConnectionPrivate
 {
diff --git a/plugins/spice/vinagre-spice-tab.c b/plugins/spice/vinagre-spice-tab.c
index 7781c98..11e3587 100644
--- a/plugins/spice/vinagre-spice-tab.c
+++ b/plugins/spice/vinagre-spice-tab.c
@@ -32,6 +32,7 @@
 #include "vinagre-spice-tab.h"
 #include "vinagre-spice-connection.h"
 #include "vinagre-spice-tunnel.h"
+#include "vinagre-utils.h"
 
 #define VINAGRE_SPICE_TAB_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), VINAGRE_TYPE_SPICE_TAB, VinagreSpiceTabPrivate))
 
diff --git a/plugins/vnc/vinagre-vnc-connection.c b/plugins/vnc/vinagre-vnc-connection.c
index 2182743..355ef54 100644
--- a/plugins/vnc/vinagre-vnc-connection.c
+++ b/plugins/vnc/vinagre-vnc-connection.c
@@ -25,6 +25,7 @@
 #include <vinagre/vinagre-cache-prefs.h>
 
 #include "vinagre-vnc-connection.h"
+#include "vinagre-utils.h"
 
 struct _VinagreVncConnectionPrivate
 {
diff --git a/plugins/vnc/vinagre-vnc-tab.c b/plugins/vnc/vinagre-vnc-tab.c
index 914cdea..5329d73 100644
--- a/plugins/vnc/vinagre-vnc-tab.c
+++ b/plugins/vnc/vinagre-vnc-tab.c
@@ -29,6 +29,7 @@
 #include "vinagre-vnc-tab.h"
 #include "vinagre-vnc-connection.h"
 #include "vinagre-vnc-tunnel.h"
+#include "vinagre-utils.h"
 
 #define VINAGRE_VNC_TAB_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), VINAGRE_TYPE_VNC_TAB, VinagreVncTabPrivate))
 
diff --git a/vinagre/vinagre-bookmarks-migration.c b/vinagre/vinagre-bookmarks-migration.c
index 3fae66b..2afe5cd 100644
--- a/vinagre/vinagre-bookmarks-migration.c
+++ b/vinagre/vinagre-bookmarks-migration.c
@@ -36,6 +36,7 @@
 #include "vinagre-bookmarks-migration.h"
 #include "vinagre-bookmarks.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 static void
 fill_xml (GSList *list, xmlTextWriter *writer)
diff --git a/vinagre/vinagre-bookmarks-ui.c b/vinagre/vinagre-bookmarks-ui.c
index 5f37aa2..0837eae 100644
--- a/vinagre/vinagre-bookmarks-ui.c
+++ b/vinagre/vinagre-bookmarks-ui.c
@@ -25,6 +25,7 @@
 #include "vinagre-bookmarks-ui.h"
 #include "vinagre-bookmarks-tree.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 static void
 control_save_button_visibility (GtkEntry *ed, GtkWidget *bt)
diff --git a/vinagre/vinagre-bookmarks.c b/vinagre/vinagre-bookmarks.c
index 968be99..b051c50 100644
--- a/vinagre/vinagre-bookmarks.c
+++ b/vinagre/vinagre-bookmarks.c
@@ -30,6 +30,7 @@
 #include "vinagre-bookmarks-migration.h"
 #include "vinagre-connection.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 struct _VinagreBookmarksPrivate
 {
diff --git a/vinagre/vinagre-cache-prefs.c b/vinagre/vinagre-cache-prefs.c
index 4330d74..407ffd8 100644
--- a/vinagre/vinagre-cache-prefs.c
+++ b/vinagre/vinagre-cache-prefs.c
@@ -21,6 +21,7 @@
 #include <config.h>
 #include <glib/gi18n.h>
 #include "vinagre-cache-prefs.h"
+#include "vinagre-utils.h"
 
 static GKeyFile *keyfile = NULL;
 static char* filename = NULL;
diff --git a/vinagre/vinagre-commands.c b/vinagre/vinagre-commands.c
index fefd40b..db91dd6 100644
--- a/vinagre/vinagre-commands.c
+++ b/vinagre/vinagre-commands.c
@@ -39,6 +39,7 @@
 #include "vinagre-cache-prefs.h"
 #include "vinagre-plugins-engine.h"
 #include "vinagre-reverse-vnc-listener-dialog.h"
+#include "vinagre-utils.h"
 
 void
 vinagre_cmd_direct_connect (VinagreConnection *conn,
diff --git a/vinagre/vinagre-connect.c b/vinagre/vinagre-connect.c
index 9b78de9..87db4b3 100644
--- a/vinagre/vinagre-connect.c
+++ b/vinagre/vinagre-connect.c
@@ -36,6 +36,7 @@
 #include "vinagre-prefs.h"
 #include "vinagre-cache-prefs.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 typedef struct {
   GtkBuilder *xml;
diff --git a/vinagre/vinagre-connection.c b/vinagre/vinagre-connection.c
index 2bb6245..e1db7bf 100644
--- a/vinagre/vinagre-connection.c
+++ b/vinagre/vinagre-connection.c
@@ -27,6 +27,7 @@
 #include "vinagre-connection.h"
 #include "vinagre-bookmarks.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 struct _VinagreConnectionPrivate
 {
diff --git a/vinagre/vinagre-notebook.c b/vinagre/vinagre-notebook.c
index e7bd82f..1af30f2 100644
--- a/vinagre/vinagre-notebook.c
+++ b/vinagre/vinagre-notebook.c
@@ -26,6 +26,7 @@
 
 #include "vinagre-dnd.h"
 #include "vinagre-prefs.h"
+#include "vinagre-utils.h"
 
 #define VINAGRE_NOTEBOOK_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), VINAGRE_TYPE_NOTEBOOK, VinagreNotebookPrivate))
 
diff --git a/vinagre/vinagre-options.c b/vinagre/vinagre-options.c
index 25a68d5..4d83ea5 100644
--- a/vinagre/vinagre-options.c
+++ b/vinagre/vinagre-options.c
@@ -23,6 +23,8 @@
 #include "vinagre-connection.h"
 #include "vinagre-window.h"
 #include "vinagre-commands.h"
+#include "vinagre-options.h"
+#include "vinagre-utils.h"
 
 const GOptionEntry all_options [] =
 {
diff --git a/vinagre/vinagre-reverse-vnc-listener-dialog.c b/vinagre/vinagre-reverse-vnc-listener-dialog.c
index 9f649b9..cd53b86 100644
--- a/vinagre/vinagre-reverse-vnc-listener-dialog.c
+++ b/vinagre/vinagre-reverse-vnc-listener-dialog.c
@@ -44,6 +44,7 @@
 #include "vinagre-prefs.h"
 #include "vinagre-reverse-vnc-listener-dialog.h"
 #include "vinagre-reverse-vnc-listener.h"
+#include "vinagre-utils.h"
 
 typedef struct
 {
diff --git a/vinagre/vinagre-reverse-vnc-listener.c b/vinagre/vinagre-reverse-vnc-listener.c
index ab314f5..54bca9c 100644
--- a/vinagre/vinagre-reverse-vnc-listener.c
+++ b/vinagre/vinagre-reverse-vnc-listener.c
@@ -37,6 +37,7 @@
 #include "vinagre-commands.h"
 #include "vinagre-reverse-vnc-listener.h"
 #include "plugins/vnc/vinagre-vnc-connection.h"
+#include "vinagre-utils.h"
 
 struct _VinagreReverseVncListenerPrivate
 {
diff --git a/vinagre/vinagre-ssh.c b/vinagre/vinagre-ssh.c
index c5b3427..d18b854 100644
--- a/vinagre/vinagre-ssh.c
+++ b/vinagre/vinagre-ssh.c
@@ -22,6 +22,7 @@
 #include <config.h>
 
 #include "vinagre-ssh.h"
+#include "vinagre-utils.h"
 #include "pty_open.h"
 
 #ifdef G_OS_WIN32
diff --git a/vinagre/vinagre-tab.c b/vinagre/vinagre-tab.c
index a53543f..0113b86 100644
--- a/vinagre/vinagre-tab.c
+++ b/vinagre/vinagre-tab.c
@@ -31,6 +31,7 @@
 #include "vinagre-prefs.h"
 #include "view/autoDrawer.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 #define VINAGRE_TAB_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), VINAGRE_TYPE_TAB, VinagreTabPrivate))
 
@@ -634,7 +635,7 @@ void
 vinagre_tab_add_view (VinagreTab *tab, GtkWidget *view)
 {
   GtkWidget *viewport;
-  GdkColor color = {0,};
+  GdkRGBA color = {0,};
 
   g_return_if_fail (VINAGRE_IS_TAB (tab));
 
@@ -643,7 +644,7 @@ vinagre_tab_add_view (VinagreTab *tab, GtkWidget *view)
 					 view);
   viewport = gtk_bin_get_child (GTK_BIN (tab->priv->scroll));
   gtk_viewport_set_shadow_type(GTK_VIEWPORT (viewport), GTK_SHADOW_NONE);
-  gtk_widget_modify_bg (viewport, GTK_STATE_NORMAL, &color);
+  gtk_widget_override_background_color (viewport, GTK_STATE_NORMAL, &color);
 }
 
 /**
diff --git a/vinagre/vinagre-tube-handler.c b/vinagre/vinagre-tube-handler.c
index 6b4d767..f40702a 100644
--- a/vinagre/vinagre-tube-handler.c
+++ b/vinagre/vinagre-tube-handler.c
@@ -39,6 +39,8 @@
 #include "vinagre-debug.h"
 #include "vinagre-protocol.h"
 #include "vinagre-plugins-engine.h"
+#include "plugins/vnc/vinagre-vnc-connection.h"
+#include "vinagre-utils.h"
 
 G_DEFINE_TYPE (VinagreTubeHandler, vinagre_tube_handler, G_TYPE_OBJECT);
 
diff --git a/vinagre/vinagre-window.c b/vinagre/vinagre-window.c
index 92c20b3..017aa58 100644
--- a/vinagre/vinagre-window.c
+++ b/vinagre/vinagre-window.c
@@ -41,6 +41,7 @@
 #include "vinagre-window-private.h"
 #include "vinagre-bookmarks-entry.h"
 #include "vinagre-plugins-engine.h"
+#include "vinagre-utils.h"
 
 #ifdef VINAGRE_HAVE_AVAHI
 #include "vinagre-mdns.h"
-- 
1.7.6.1

